src/NUnitConsole/nunit3-console.tests/CommandLineTests.cs

```diff
@@ -609,164 +609,133 @@ public void ShouldNotFailOnEmptyLine()
         public void ParameterWithMissingNameIsInvalid()
         {
-            var options = ConsoleMocks.Options("--testparam:=5");
+            var options = ConsoleMocks.Options("--param:=5");
             Assert.That(options.ErrorMessages.Count, Is.EqualTo(1));
         }
 
         [Test]
         public void ParameterWithMissingValueIsInvalid()
         {
-            var options = ConsoleMocks.Options("--testparam:X=");
+            var options = ConsoleMocks.Options("--param:X=");
             Assert.That(options.ErrorMessages.Count, Is.EqualTo(1));
         }
 
         [Test]
-        public void LeadingWhitespaceIsPreservedInParameterName()
+        public void LeadingWhitespaceIsRemovedInParameterName()
         {
             // Command line examples to get in this scenario:
-            // --testparams:"  X"=5
-            // --testparams:"  X=5"
-            // "--testparams:  X=5"
+            // --param:"  X"=5
+            // --param:"  X=5"
+            // "--param:  X=5"
 
-            var options = ConsoleMocks.Options("--testparam:  X=5");
-            Assert.That(options.TestParameters, Is.EqualTo(new Dictionary<string, string> { ["  X"] = "5" }));
+            var options = ConsoleMocks.Options("--param:  X=5");
+            Assert.That(options.TestParameters, Is.EqualTo(new Dictionary<string, string> { ["X"] = "5" }));
         }
 
         [Test]
-        public void TrailingWhitespaceIsPreservedInParameterName()
+        public void TrailingWhitespaceIsRemovedInParameterName()
         {
             // Command line examples to get in this scenario:
-            // --testparams:"X  "=5
-            // --testparams:"X  =5"
-            // "--testparams:X  =5"
+            // --param:"X  "=5
+            // --param:"X  =5"
+            // "--param:X  =5"
 
-            var options = ConsoleMocks.Options("--testparam:X  =5");
-            Assert.That(options.TestParameters, Is.EqualTo(new Dictionary<string, string> { ["X  "] = "5" }));
+            var options = ConsoleMocks.Options("--param:X  =5");
+            Assert.That(options.TestParameters, Is.EqualTo(new Dictionary<string, string> { ["X"] = "5" }));
         }
 
         [Test]
-        public void WhitespaceIsPermittedAsParameterName()
+        public void WhitespaceIsNotPermittedAsParameterName()
         {
             // Command line examples to get in this scenario:
             // --testparams:"  "=5
             // --testparams:"  =5"
             // "--testparams:  =5"
 
-            var options = ConsoleMocks.Options("--testparam:  =5");
-            Assert.That(options.TestParameters, Is.EqualTo(new Dictionary<string, string> { ["  "] = "5" }));
+            var options = ConsoleMocks.Options("--param:  =5");
+            Assert.That(options.ErrorMessages.Count, Is.EqualTo(1));
         }
 
         [Test]
-        public void LeadingWhitespaceIsPreservedInParameterValue()
+        public void LeadingWhitespaceIsRemovedInParameterValue()
```

Comment:
Why do we want this behavior to change?

Reply:
It seemed more like what one would expect. If you type `X = 5`, don't you expect X to be 5 rather than " 5" ?

Ideally I would have  wanted your first example to work differently from the other two, but of course we get the same thing in all three cases after Windows command-line processing.

Reply:
I'm not totally in love with this solution either. :-(

Reply:
The problem is, what if someone types `X=" 5"` at the command line? When that gets turned into `string[]` using standard Windows argv rules, the string we get in the array is `"X= 5"`.

How then should people put leading or trailing spaces in test parameters if they want to?

Reply:
(There's no way to distinguish whether `"X= 5"` or `X=" 5"` was typed by looking in `string[] args`.)

Reply:
Also, if someone typed `X = 5` at the command line without quotes, `string[] args` would contain the three elements `X`, `=`, `5`.

Reply:
So... thinking about it, all of these same considerations apply in spades to the `--where` option, because there are even more possibilities as to how and where you use quotes on the command-line. Mostly, I've dealt with the problems people bring up by saying: "don't do that, do this." Can we do that here? The practical rules would be...

1.  Use =, : or space after --param
2.  Avoid spaces in the argument that follows, but
3. If you must have spaces, put the whole following argument in quotes
4. If the value part of the argument has spaces, use single or escaped double quotes around that part only.

So
 --param X=5
 --param="X = 5"
 -p:"X='Something with spaces'" (OR "\"Something with spaces\""

Obviously, that would all have to be documented.

OTOH, we could go back to the existing behavior, which you documented in tests, but then we would be inconsistent with how --where is processed and up to now that has been OK.

Reply:
The only downside to changing it is that wrappers of NUnit such as Cake and NUKE will not be able to share logic for escaping parameters between v3 and v4.

Reply:
> If the value part of the argument has spaces, use single or escaped double quotes around that part only.

Just as a reminder, with argv parsing, these three things all result in the string `--param=X= 5` being in `string[] args`:

- `"--param=X= 5"`
- `--param="X= 5"`
- `--param=X=" 5"`

We can't tell them apart. So we will either have to require single quotes, or never trim the value part. (Similar for the name part, but how likely is that to actually want whitespace around it?)

Reply:
Yes, that's my point exactly. However, my conclusion from the same facts seems to be different. (No surprise)

If we __could__ tell them apart, I'd want the third one to work differently from the first two. But we can't and I hope our users know that as well. (That last part is a bit shaky, since some of our users these days are not programmers.)

In any case, I'd like us to be consistent for all the options that take name/value pairs.

So which side are you coming down on here? Are you saying you're OK with trimming names but not values?